storage/csv/datasource.go

```diff
@@ -58,9 +62,35 @@ func (ds *DataSource) Get(variables octosql.Variables) (execution.RecordStream,
 	r := csv.NewReader(file)
 	r.TrimLeadingSpace = true
 
-	columns, err := r.Read()
-	if err != nil {
-		return nil, errors.Wrap(err, "couldn't read column names")
+	var columns []string
+
+	/* if the files has column names then it's the first row */
+	if ds.hasColumnNames {
+		columns, err = r.Read()
+		if err != nil {
+			return nil, errors.Wrap(err, "couldn't read column names")
+		}
+	} else { /* otherwise we must determine how many rows there are and create columns col1, col2... */
+		lengthReader := csv.NewReader(file)
+		lengthReader.TrimLeadingSpace = true
+
+		firstRow, err := lengthReader.Read()
+		if err != nil {
+			return nil, errors.Wrap(err, "couldn't read first row to determine number of columns")
+		}
+
+		columnCount := len(firstRow)
+
+		columns = make([]string, columnCount)
+
+		for i := 0; i < columnCount; i++ {
+			columns[i] = fmt.Sprintf("col%d", i+1)
+		}
+
+		_, err = file.Seek(0, 0) /* we must return to the beginning of file */
```

Comment:
Why? The first row was the header row, we don't want it anymore

Reply:
It's actually the opposite. There is no header row, so we read the first one, to determine how many columns there are (to create column names col1, col2...). Alternatively the first row could be passed to the stream as a []string, same as in Excel. Don't know which method is better tough

Reply:
As discussed in private, I think it's better to read and save the column list on first record read.